netx/net/sourceforge/jnlp/controlpanel/JVMPanel.java

```diff
@@ -234,7 +235,9 @@ public static JvmValidationResult validateJvm(String cmd) {
             validationResult += "<span color=\"red\">" + Translator.R("CPJVMnotDir") + "</span><br />";
             latestOne = JvmValidationResult.STATE.NOT_DIR;
         }
-        File javaFile = new File(cmd + File.separator + "bin" + File.separator + "java");
+        File javaFile = new File(cmd + File.separator + "bin" + 
+                                       File.separator + "java" + 
+                                       (JNLPRuntime.isWindows() ? ".exe" : ""));
```

Comment:
Isn't this more spread? If so, maybe it can get its own function?

Reply:
Sorry i don't understand wath you mean with "Isn't this more spread?" I am not an English native speaker.

Reply:
Nor am I. I suspect that java part could forget .exe  on more places. So maybe it would be good to ahve proper metho to appned .exe if necessary.

Reply:
All others I know are in the rust sources.